counterpartylib/lib/messages/issuance.py

```diff
@@ -186,7 +233,17 @@ def compose (db, source, transfer_destination, asset, quantity, divisible, descr
         call_price = 0.0
     cursor.close()
 
-    call_date, call_price, problems, fee, description, divisible, reissuance = validate(db, source, transfer_destination, asset, quantity, divisible, callable_, call_date, call_price, description, util.CURRENT_BLOCK_INDEX)
+    # check subasset
+    subasset_parent = None
+    subasset_longname = None
+    if util.enabled('subassets'): # Protocol change.
+        subasset_parent, subasset_longname = util.parse_subasset_from_asset_name(asset)
+        if subasset_longname is not None:
+            # generate a random numeric asset id which maps to this subasset
+            asset = util.generate_random_asset()
+            description = description + ';;l' + subasset_longname
```

Comment:
actually, probably keep it as-is. we use magic numbers/letters enough in this file...when they're this short it's a matter of taste, and I'd rather code for readability rather than purism